app/controllers/bookmarks_controller.rb

```diff
@@ -138,7 +138,7 @@ def index
           search = BookmarkSearchForm.new(show_private: false, show_restricted: false, sort_column: 'created_at')
           results = search.search_results
           flash_search_warnings(results)
-          @bookmarks = results.to_a
+          results.to_a
```

Comment:
I don't understand why this was changed, however if the tests continue to pass I'm happy with it.

Reply:
I kept on looking at this bit of code when I was wondering why the memcached tests were failing and the assignment at the end of the cache block is just wrong 

The old code is

>        @bookmarks = Rails.cache.fetch("bookmarks/index/latest/v2_true", expires_in: ArchiveConfig.SECONDS_UNTIL_BOOKMARK_INDEX_EXPIRE.seconds) do
>           search = BookmarkSearchForm.new(show_private: false, show_restricted: false, sort_column: 'created_at')
>           results = search.search_results
>           flash_search_warnings(results)
>            @bookmarks =results.to_a
>         end

Rails.cache.fetch evaluates the block on a cache miss, at the end of this block we assign @bookmarks  ( which is a side effect that will not happen on a cache hit ), the result of the Rails.cache.fetch is the last evaulation, which is the result of the assignment which is the value being assigned. That value is then assigned to @bookmarks again.

Basically once I understood the code it made me cry and I couldn't help but want to make it simpler. Its not related to this and I should have done it in another merge request but they are identical in execution but this makes me want to cry less.

Reply:
:joy: Thanks for explaining! That makes sense. Happy to include it in this PR if you are, I'll just leave it as a separate commit so it's clear it's not related to the other changes we're making